src/UnoFlipModel.java

```diff
@@ -12,68 +12,330 @@
  * Date: 2023-10-20
  */
 public class UnoFlipModel {
-    private ArrayList<Player> players; //arraylist of players
+
+    private boolean turnFinished;
+    private boolean skipTurn;
+    private boolean challenge; //true if next player wants to challenge a wild draw 2, false if they do not
+    private boolean dontAskChallenge; //true if game does not want give player challenge prompt, false if game want to give player challenge prompt
     private boolean turnDirection; //true is clockwise(1->2->3->4), false is counterclockwise(1->4->3->2)
-    private int currentTurn; // 0 is player 1, 1 is player 2, 2 is player 3, 3 is player 4
-    private int nextPlayerIndex; //index of next player
-    private Deck deck; //deck that will be used for the game
-    private Card.Colour currentColour; //current colour of the game
+    private int numPlayers;
+    private int chosenCardIndex;
+    private int currentTurn; // 0-indexed (ex. 0 is player 1, 1 is player 2, 2 is player 3, 3 is player 4)
+    private int nextPlayerIndex;
+    private String status; //indicate the status for which the view will update to
+    private Deck deck;
+    private Card.Colour currentColour;
+    private Card.Rank currentRank;
+    private Card topCard;
+    private List<UnoFlipView> views;
+    private ArrayList<TurnSequence> turnSeqs;
+    private ArrayList<Player> players;
+
+    public static final int DRAW_ONE_BUTTON = -1; // draw one button indicator
```

Comment:
there should be no mention of buttons here. this is the model, i have mentioned a few times in this review not to couple the model and the view unnecessarily